.github/workflows/test-upstream.yaml

```diff
@@ -24,6 +24,9 @@ jobs:
           - python-version: "3.10"
             pydantic-version: "1"
```

Comment:
I know it's not even germane to the changes here but I recently simplified the testing matrix in the `linkml` repo (https://github.com/linkml/linkml/pull/2041). It might be worth using the same matrix here. I wouldn't lose sleep at night if we didn't do a Pydantic v1 test in this upstream workflow.

Reply:
ya ya good call, i wrote this before that got merged, and we were definitely thinking the same thing ;). so would b good to match.